test/sanitize_json.d

```diff
@@ -39,25 +39,61 @@ int main(string[] args)
     }
 
     auto json = parseJSON(readText(inFilename));
-    sanitize(json.array);
+    sanitize(json);
 
     outFile.write(json.toJSON(true));
     return 0;
 }
 
-void sanitize(JSONValue[] rootArray)
+void sanitize(JSONValue root)
 {
-    foreach (ref obj; rootArray)
+    if (root.type == JSON_TYPE.ARRAY)
     {
-        auto kind = obj.object["kind"].str;
-        if (kind == "compilerInfo")
-            sanitizeCompilerInfo(obj.object);
-        else if (kind == "buildInfo")
-            sanitizeBuildInfo(obj.object);
-        else if(kind == "module")
-            sanitizeSyntaxNode(obj);
-        else if(kind == "semantics")
-            sanitizeSemantics(obj.object);
+        foreach (ref obj; root.array)
+        {
+            auto kind = obj.object["kind"].str;
+            if (kind == "compilerInfo")
```

Comment:
DRY + type safety:
s/"compilerInfo"/JsonField.compilerInfo.stringof/
+ everywhere relevant

Reply:
This is the test suite, not dmd.  It doesn't have access to the `JsonField` type defined in the compiler.

Reply:
`import dmd.json : JsonField;` ?

Reply:
Yeah, the testsuite shouldn't (directly) depend on what it is testing.